### ISSUE TITLE
`CUDNN_STATUS_BAD_PARAM` in BatchNorm backward

### ISSUE THREAD

**Issue by alykhantejani** [pytorch/pytorch#1866]
```Python
import torch
from torch import nn
import torch.nn.functional as F
from torch.autograd import Variable

bn = nn.BatchNorm1d(10)
bn = bn.cuda()
input = Variable(torch.Tensor(4, 10).cuda())

o = bn(input)
o.sum().backward()
```

This throws a `CUDNN_STATUS_BAD_PARAM` on the `backward` call 

Its fine in v0.12 and is fine in 5c7453447f54721f7d08952a08d541606e1b3a68 but broken in c573d53939ea2be6f660934738811a521470ab08

Still trying to track it down but compiling takes ages

**Comment by alykhantejani** [comment]
Ok, I found the problem (with the help of @fmassa). the `sum` has an `expand` in the backwards which makes the `grad_output` into the batch norm non-contiguous. This is related to issue #603 as there are no contiguous checks in the cudnn path (and no calls to make the `grad_output` contiguous). 

I'll send a PR for this, this week.
